### ISSUE TITLE
Status Checker and Status Button components don't update when connection lost to server

### ISSUE THREAD

**Issue by Markavian** [connected-web/product-monitor#19]
They fail to update their status correctly when a fail condition is returned from the server.

Improvements could be made to Web Component for this, but I've identified an improvement in the preRender step for these components:

``` js
    this.statusCode = (this.dataSourceError === false) ? this.statusCode : 503;
    this.alertType = this.selectAlertType(this.statusCode);
```

Because most of render state is based on the `this.selectAlertType` and `this.selectMessage` render functions, by patching the statusCode in the case of an error, it fixes the overall state of the component.

``` js
<script type="text/javascript">
$(function() {

  var StatusChecker = Component.configure("status-checker");

  StatusChecker.prototype.selectMessage = function(statusCode) {
    return ({
      "0": 'Checking server',
      "200": 'URL Seems OK',
      "404": "File not found",
      "500": "Server responded with an error: " + this.dataSourceData,
      "502": "The server encountered a problem: " + this.dataSourceData,
      "503": "Server is unavailable"
    })[statusCode] || this.dataSourceStatus;
  }

  StatusChecker.prototype.selectAlertType = function(statusCode) {
    return ({
      "0": 'info',
      "200": 'success',
      "404": 'danger',
      "500": 'warning',
      "502": 'warning'
      "503": 'danger'
    })[this.statusCode] || 'danger';
  }

  StatusChecker.prototype.preRenderStep(function() {
    this.statusCode = (this.dataSourceError === false) ? this.statusCode : 503;
    this.alertType = this.selectAlertType(this.statusCode);
    this.message = this.selectMessage(this.statusCode);
  });

});
</script>
```

**Comment by subsidel** [comment]
Can this be closed after 24b63aa?

**Comment by Markavian** [comment]
Yes, closing now.
